### ISSUE TITLE
OData Batches and Content-ID

### ISSUE THREAD

**Issue by emragins** [iwate/ODataHttpClient#12]
I'd like to use the Content-ID from the batches coming back so that we can correlate a failed response with the corresponding request.  

Using Fiddler, the target application provides this as the raw text response for a batch with 5 items (with the 4th set to fail):

```
--batchresponse_2bbf2f24-1194-4138-92f0-527b237a68a5
Content-Type: multipart/mixed; boundary=changesetresponse_869711e5-41d1-4301-8510-3eb4b6a0c9b8

--changesetresponse_869711e5-41d1-4301-8510-3eb4b6a0c9b8
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-ID: 4

HTTP/1.1 500 Internal Server Error
Content-Type: application/json; odata.metadata=minimal
OData-Version: 4.0

{
  "error":{
```

I've pulled the source and it looks like in `ParseMultiAsync(...)` it's available in `content.Headers`  in this block:
```
if (content.Headers.ContentType.MediaType == "application/http")
                {
                    if (!content.Headers.ContentType.Parameters.Contains(_responseMsgType))
                        content.Headers.ContentType.Parameters.Add(_responseMsgType);
                    
                    var part = await content.ReadAsHttpResponseMessageAsync();

                    //headers.Add("Content-ID", content.Headers.GetValues("Content-ID"));
                    result.Add(await ParseAsync(part.StatusCode, part.Content, headers));
                }
```

However, it's not in the `headers` that are added into the `Response`.

What is the recommended approach for tackle this?  I'm happy to modify the source code and submit a pull-request, but I also want to make sure to stay true to the minimalist vision of this library.

If there's some way to solve that doesn't involve modifying source, that would be really helpful to know.

Thank you.

**Comment by iwate** [comment]
https://github.com/iwate/ODataHttpClient/blob/eabafc5d5f3b1056064ff5f9a3242ae9216b93e5/src/ODataHttpClient/ODataClient.cs#L70

It seems a bug 🐛 

My plan for fix this bug is following.

```
result.Add(await ParseAsync(part.StatusCode, part.Content, part.Headers));
```

@emragins  I think it is perfect for your issue. How is this?

**Comment by emragins** [comment]
Thanks for responding to this.  It doesn't look like that works, though.  `part` doesn't have the correct headers, but `content.Headers` does.  I took a screenshot demonstrating this:

![image](https://user-images.githubusercontent.com/1704032/72655233-7addb480-3948-11ea-81fd-1f875fffb1df.png)

**Comment by iwate** [comment]
I check my tests: SenarioTestCreateProductWithCategory(V3), SenarioV4Test.CreateProductWithCategory(V4).

V3 has Content-ID in `part`
```
HTTP/1.1 202 Accepted
Cache-Control: no-cache
Content-Length: 3744
Content-Type: multipart/mixed; boundary=batchresponse_052ebcef-1dab-4abe-9dd5-312b8facc977
Server: Microsoft-IIS/10.0
X-Content-Type-Options: nosniff
DataServiceVersion: 1.0;
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET
Access-Control-Allow-Headers: Accept, Origin, Content-Type, MaxDataServiceVersion
Access-Control-Expose-Headers: DataServiceVersion
Date: Mon, 20 Jan 2020 03:54:16 GMT

--batchresponse_052ebcef-1dab-4abe-9dd5-312b8facc977
Content-Type: multipart/mixed; boundary=changesetresponse_0959cb19-74af-4bf3-9676-a69e01ea8f36

--changesetresponse_0959cb19-74af-4bf3-9676-a69e01ea8f36
Content-Type: application/http
Content-Transfer-Encoding: binary

HTTP/1.1 201 Created
DataServiceVersion: 3.0;
Content-Type: application/json;odata=minimalmetadata;streaming=true;charset=utf-8
Content-ID: 1
X-Content-Type-Options: nosniff
Cache-Control: no-cache
Location: https://services.odata.org/V3/(S(ni3esrvxcoxxfea2kdchzo0o))/OData/OData.svc/Products(100)

{"odata.metadata":"https://services.odata.org/V3/(S(ni3esrvxcoxxfea2kdchzo0o))/OData/OData.svc/$metadata#Products/@Element","ID":100,"Name":"Test Product","Description":"Test Item","ReleaseDate":"2020-01-20T03:54:16.489406+00:00","DiscontinuedDate":null,"Rating":1,"Price":1.0}
```

But V4 has Content-Id in `content`

```
HTTP/1.1 202 Accepted
Cache-Control: no-cache
Content-Length: 3793
Content-Type: multipart/mixed; boundary=batchresponse_da688dea-11c9-4523-8003-fa4c82d37605
Server: Microsoft-IIS/10.0
X-Content-Type-Options: nosniff
OData-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET
Access-Control-Allow-Headers: Accept, Origin, Content-Type, MaxDataServiceVersion
Access-Control-Expose-Headers: DataServiceVersion
Date: Mon, 20 Jan 2020 03:46:48 GMT

--batchresponse_da688dea-11c9-4523-8003-fa4c82d37605
Content-Type: multipart/mixed; boundary=changesetresponse_902fa22b-6b70-4839-8759-b2523b9c2e5e

--changesetresponse_902fa22b-6b70-4839-8759-b2523b9c2e5e
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-ID: 1

HTTP/1.1 201 Created
OData-Version: 4.0;
Content-Type: application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8
X-Content-Type-Options: nosniff
Cache-Control: no-cache
Location: https://services.odata.org/V4/OData/(S(sdwxdzr12jf3tofcuinigded))/OData.svc/Products(100)

{"@odata.context":"https://services.odata.org/V4/OData/(S(sdwxdzr12jf3tofcuinigded))/OData.svc/$metadata#Products/$entity","ID":100,"Name":"Test Product","Description":"Test Item","ReleaseDate":"2020-01-20T12:46:48.3055565+09:00","DiscontinuedDate":null,"Rating":1,"Price":1.0}
```

**Comment by iwate** [comment]
@emragins 

I update code. https://github.com/iwate/ODataHttpClient/commit/5be2033e0ef359c4a0e0810db0ed4f421b180afd

if content has Content-ID and part does not have Content-ID, copy to part.
How is this?

**Comment by emragins** [comment]
Good catch with the version difference.  That looks like it should work!

**Comment by iwate** [comment]
Fixed version (v1.6.8) is released
https://www.nuget.org/packages/ODataHttpClient/
